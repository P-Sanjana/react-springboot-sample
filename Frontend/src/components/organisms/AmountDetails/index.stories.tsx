import React from "react";
import AmountDetails from ".";
export default {
  title: "organisms/AmountDetails",
  component: AmountDetails,
};
import { Currency, Wallet } from "../../../types/responseTypes";
export const selectedCrypto = {
  id: 1,
  currencyType: "CRYPTO",
  name: "Bitcoin",
  code: "BTC",
  about: "ABout",
  logoUrl: "img",
  color: "#FFFFFF",
  price: 1002434,
  marketCap: 673434,
  marketCapShort: "3748",
  vol24h: "74733",
  circulatingSupply: "74837442",
  watch: true,
  changeRate: 2.3,
};
const balance = {
  id: 1,
  userId: 1,
  currencyId: 1,
  currencyType: "CRYPTO",
  amount: 378438,
};

export const amountDetails = () => (
  <AmountDetails
    amountEventHandler={() => console.log("amount entered")}
    selectedCrypto={selectedCrypto}
    amountEntered={34000.0}
    convertedCoinHandler={() => console.log("converted")}
    convertedCoin={0.023451}
    buttonTitle="Buy max"
    totalBal={2844}
    cryptoBal={balance}
    error=""
    setErrorHandler={() => console.log("error message")}
  />
);
