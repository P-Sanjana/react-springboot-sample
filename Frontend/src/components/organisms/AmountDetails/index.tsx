import {
  Card,
  makeStyles,
  Slider,
  Typography,
  withStyles,
  Button,
  Box,
  InputBase,
  InputAdornment,
} from "@material-ui/core";
import React, { useEffect } from "react";
import theme from "../../../theme/theme";
import {
  amountDetails,
  dollarSign,
  one,
  equalToSign,
  errorMessage,
} from "../../../utils/constants";
import localeUtils from "../../../utils/locale-utils";
import { serverCryptoProps } from "../SelectionCard";
import { totalBalType } from "../PaymentMethod";
import { Currency, Wallet } from "../../../types/responseTypes";
const styles = makeStyles(() => ({
  root: {
    display: "flex",
    flexDirection: "column",
    padding: theme.spacing(6),
    width: "100%",
    border: `1px solid ${theme.palette.grey["100"]}`,
    boxSizing: "border-box",
    borderRadius: theme.spacing(1),
  },
  firstCard: {
    paddingTop: "2",
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    padding: theme.spacing(4),
    border: `1px solid ${theme.palette.grey["100"]}`,
    borderRadius: theme.spacing(1),
    marginTop: theme.spacing(3),
  },
  sliderCard: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    height: "88px",
    paddingLeft: theme.spacing(5),
  },
  secondCard: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    padding: theme.spacing(4),
    border: `1px solid ${theme.palette.grey["100"]}`,
    borderRadius: theme.spacing(1),
  },

  slider: {
    height: "88px",
    backgroundColor: theme.palette.text.disabled,
    border: `2px solid ${theme.palette.text.disabled}`,
  },
  bitcoinValue: {
    color: theme.palette.text.hint,
  },
  button: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    height: "42px",
    border: `1px solid ${theme.palette.primary["500"]}`,
    borderRadius: theme.spacing(1),
  },
  BTCText: {
    color: theme.palette.text.hint,
  },
}));

export type amountEnteredEventHandler = (amount: number) => void;
export type convertedCoinEventHandler = (coinAmount: number) => void;
export type errorMessageHandler = (errorMsg: string) => void;

export interface AmountDetailsProps {
  amountEventHandler: amountEnteredEventHandler;
  selectedCrypto: Currency | undefined;
  amountEntered: number;
  buttonTitle: string;
  convertedCoin: number | undefined;
  convertedCoinHandler: convertedCoinEventHandler;
  totalBal: number | undefined;
  error: string | undefined;
  setErrorHandler: errorMessageHandler;
  cryptoBal: Wallet | undefined;
}

const CustomSlider = withStyles({
  thumb: {
    color: theme.palette.text.disabled,
    border: `1px solid ${theme.palette.text.disabled}`,
  },
  rail: {
    color: theme.palette.text.disabled,
    background: theme.palette.text.disabled,
    border: `1px solid ${theme.palette.text.disabled}`,
  },
  track: {
    color: theme.palette.text.disabled,
    background: theme.palette.text.disabled,
    border: `1px solid ${theme.palette.text.disabled}`,
  },
})(Slider);
const AmountDetails: React.FC<AmountDetailsProps> = ({
  amountEntered,
  amountEventHandler,
  selectedCrypto,
  buttonTitle,
  convertedCoin,
  convertedCoinHandler,
  totalBal,
  error,
  setErrorHandler,
  cryptoBal,
}) => {
  const style = styles();
  let balance: number;
  let cash: number;
  useEffect(() => {
    if (isNaN(amountEntered)) {
      convertedCoinHandler(0);
    }
    console.log(totalBal);
    if (amountEntered && selectedCrypto && cryptoBal) {
      const converted = amountEntered / selectedCrypto.price;
      if (
        convertedCoin &&
        convertedCoin > cryptoBal.amount &&
        buttonTitle === "Sell max"
      ) {
        setErrorHandler(errorMessage);
      } else if (
        totalBal &&
        buttonTitle === "Buy max" &&
        amountEntered > totalBal
      ) {
        setErrorHandler(errorMessage);
      } else {
        setErrorHandler("");
      }
      convertedCoinHandler(converted);
    }
  }, [amountEntered, convertedCoin]);
  return (
    <>
      <Card className={style.root} variant="outlined">
        <Typography variant="body1">{amountDetails}</Typography>
        <Card className={style.firstCard} variant="outlined">
          <Box display="flex" flexDirection="column">
            <InputBase
              placeholder="Enter Amount"
              value={amountEntered ? amountEntered : ""}
              onChange={(event) =>
                amountEventHandler(parseInt(event.target.value))
              }
              startAdornment={
                <InputAdornment position="start">
                  <Typography variant="subtitle1">{dollarSign}</Typography>
                </InputAdornment>
              }
            />
            {error && (
              <Typography
                variant="overline"
                style={{ color: theme.palette.error.main }}
              >
                {error}
              </Typography>
            )}
          </Box>
          <Button className={style.button} variant="outlined" color="primary">
            <Typography>{buttonTitle}</Typography>
          </Button>
        </Card>
        <Box className={style.sliderCard}>
          <CustomSlider
            orientation="vertical"
            defaultValue={55}
            min={10}
            max={100}
            step={10}
            track={false}
            disabled
          />
          <Typography className={style.bitcoinValue} variant="caption">
            {one}
            {selectedCrypto?.code}
            {equalToSign}
            {selectedCrypto && localeUtils.formatCurr(selectedCrypto.price)}
          </Typography>
        </Box>
        <Card className={style.secondCard} variant="outlined">
          <Typography variant="subtitle1">
            {convertedCoin ? convertedCoin : 0}
          </Typography>
          <Typography className={style.BTCText} variant="body1">
            {selectedCrypto?.code}
          </Typography>
        </Card>
      </Card>
    </>
  );
};

export default AmountDetails;
