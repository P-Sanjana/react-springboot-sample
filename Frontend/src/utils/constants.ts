import {
  CryptoProps,
  serverCryptoProps,
} from "../components/organisms/SelectionCard/index";
import Bitcoin from "../assets/coinImage//bitcoin.png";
import Ethereum from "../assets/coinImage/ethereum.png";
import Dogecoin from "../assets/coinImage/Dogecoin.png";
import Binance from "../assets/coinImage/binance.png";
import Cardano from "../assets/coinImage/Cardano.png";
import Polkadot from "../assets/coinImage/Polkadot.png";
import Tether from "../assets/coinImage/tether.png";
import XRP from "../assets/coinImage/xrp.jpeg";
import USDCoin from "../assets/coinImage/usdcoin.jpeg";
import { DropDownItem } from "../components/molecules/DropDown/DropDown";
export const test: string[] = ["hello", "bye"];
export const timePeriods: string[] = ["1H", "24H", "1W", "1M", "1Y", "ALL"];
export const aboutCrypto: string[] = [
  "About",
  "Resources",
  "Official Website",
  "White Paper",
];
export const addedToWatchList = "ADDED TO WATCHLIST";
export const checkoutList: string[] = [
  "Purchase is completed, please check your balance in your crypto wallet",
  "Buy Crypto",
  "Go To Wallet",
  "Sell is completed, please check your balance in your Rupee coin",
  "SELL CRYPTO",
  "GO TO USD COIN",
];
export const dashboard = "Dashboard";
export const trade = "Detail";
export const profilePicUrl =
  "https://s3-alpha-sig.figma.com/img/973c/8b4c/05e06771ca57ec378072223b93fc6007?Expires=1635724800&Signature=LoSuQuAoiSGeFs04cT-kZICbfCNVFSd~jnfp6ffpCUDjKiWkq9dvi61je-5x7~toH3IMbNrCQrfctP-mrlVh~H7iAF-UlMNO66--TqlXExfb91MLGLnuK8O5HONWoVL0cT9Zxqx0iBMifwRo8oY~3f45YVs52Kw2FH8LwZxbYDhZCAev85Kqk7yofawgrWgZoYQJTaS~Re4TV09PF56F4wCcbZm81XaIgCbPDF2vZg~ybhPfMAeUf18TseFlsLPU1tbCigTOF3rPjeU5bSQvaBwJrDSGNUkeEdHxpfqfV4CihwWdxdXms67T5IcnBDlMR~AzVGJL~KIiu4lLZ~Fh5g__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA";
export const profilePicName = "John Deere";
export const one = "1";
export const dash = "-";
export const cardWithGraph: string[] = ["24 h"];
export const totalBalance = "Total Balance";
export const btc = "BTC";
export const cash = "Cash";
export const usdCoin = "USD Coin";
export const defaul = "Default";
export const from = "From";
export const recentTransactions = "Recent transactions";
export const viewAll = "View All";
export const myPortfolio = ["My portfolio", "Total Balance"];
export const currentValue = "Current Value";
export const myWallets = "My wallets";
export const chooseCrypto = "Choose crypto";
export const watchlist = "Watchlist";
export const discoverAssets = "Discover assets";
export const edit = "EDIT";
export const wallet = "Wallet";
export const OrderSummaryConsts: string[] = [
  "You are ",
  "transaction fee",
  "Total",
  "$",
  "Paying through",
  "Visa credit ...8845",
  "Delivery fees",
  "0.001",
  "Deposit to",
  "Wallet",
  "Payment method",
  "Rupee Coin",
];
export const amountDetails = "Amount details";
export const dollarSign = "$";
export const rupeeSign = "â‚¹";
export const equalToSign = "=";
export const buttonTextBuy = "BUY";
export const buttonTextSell = "SELL";
export const marketcap = "Market cap";
export const vol24H = "Vol.24H";
export const circulatingsupply = "Circulating supply";
export const cashDeposit = "CASH DEPOSIT";
export const withdraw = "WITHDRAWAL";
export const deliveryFee = [
  "Select speed delivery",
  "Instant : 2-5 min",
  "Transaction fees : 0.001 BTC",
  "Instant",
  ": 2-5 minutes",
  "Delivery fees : 0.001 BTC",
  "Faster",
  ": 4 hours",
  "Delivery fees : 0.0001 BTC",
  "Fast",
  ": 120 hours",
  "Delivery fees : 0.00001 BTC",
  "None",
];
export const cryptos: Array<CryptoProps> = [
  {
    currency: "Bitcoin",
    icon: Bitcoin,
  },
  {
    currency: "Ethereum",
    icon: Ethereum,
  },
  {
    currency: "Binance",
    icon: Binance,
  },
  {
    currency: "Tether",
    icon: Tether,
  },
  {
    currency: "Cardano",
    icon: Cardano,
  },
  {
    currency: "XRP",
    icon: XRP,
  },
  {
    currency: "Dogecoin",
    icon: Dogecoin,
  },
  {
    currency: "Polkadot",
    icon: Polkadot,
  },
  {
    currency: "USDCoin",
    icon: USDCoin,
  },
];
export const portfolioValue = [
  "My portfolio value",
  "Total Investment",
  "Bitcoin",
  "$ 11,900,204",
  "$ 34,000",
];
export const dashboardpage = [
  "https://s3-alpha-sig.figma.com/img/973c/8b4c/05e06771ca57ec378072223b93fc6007?Expires=1635724800&Signature=LoSuQuAoiSGeFs04cT-kZICbfCNVFSd~jnfp6ffpCUDjKiWkq9dvi61je-5x7~toH3IMbNrCQrfctP-mrlVh~H7iAF-UlMNO66--TqlXExfb91MLGLnuK8O5HONWoVL0cT9Zxqx0iBMifwRo8oY~3f45YVs52Kw2FH8LwZxbYDhZCAev85Kqk7yofawgrWgZoYQJTaS~Re4TV09PF56F4wCcbZm81XaIgCbPDF2vZg~ybhPfMAeUf18TseFlsLPU1tbCigTOF3rPjeU5bSQvaBwJrDSGNUkeEdHxpfqfV4CihwWdxdXms67T5IcnBDlMR~AzVGJL~KIiu4lLZ~Fh5g__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
];
export const selectedCrypto: serverCryptoProps = {
  currency: "Bitcoin",
  price: "3406069.54",
  symbol: "BTC",
};
export const balance = [
  {
    id: 1,
    currency: "USD",
    wallet: "12000",
  },
  {
    id: 2,
    currency: "Bitcoin",
    wallet: "0.0022",
  },
];
export const usdCoinCash = "USD Coin (Cash)";

export const textSearchAllAssets = "Search all assets";
export const timePeriodsSelections: DropDownItem[] = [
  { key: "1H", value: "1h" },
  { key: "24H", value: "24h" },
  { key: "1W", value: "1w" },
  { key: "1M", value: "1m" },
  { key: "1Y", value: "1y" },
];
export const timePeriodsSelectionDefault = "24h";
export const assetsSelections: DropDownItem[] = [
  { key: "ALL", value: "All assets" },
];
export const assetsSelectionDefault = "All assets";
export const totalInvestmentGains = 0.48;
export const participant = "Bob";
export const errorMessage = "Entered amount should not be greater than balance";
