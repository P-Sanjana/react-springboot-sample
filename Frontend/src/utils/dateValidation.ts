import moment from "moment";
export const validateDate = (d: Date): boolean => {
  const date = moment(`${d.getFullYear()}-${d.getMonth() + 1}-${d.getDate()}`);
  return date.isValid();
};
