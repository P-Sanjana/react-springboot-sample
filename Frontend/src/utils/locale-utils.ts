//Get these values from environmantal variable
const CURRENCY_SYMBOL = "$";
const LOCALE = "en-US";

const localeUtils = {
  formatCurr: (n: number): string => {
    if (!n) {
      return `${CURRENCY_SYMBOL}0`;
    }
    return `${CURRENCY_SYMBOL}${n.toLocaleString(LOCALE)}`;
  },

  formatNumber: (n: number): string => {
    if (!n) {
      return "0";
    }
    return n.toLocaleString(LOCALE);
  },
  formatDatetime24h: (d: Date): string => {
    return d.toLocaleString(LOCALE, {
      day: "2-digit",
      month: "short",
      year: "numeric",
      hour: "2-digit",
      minute: "2-digit",
      hourCycle: "h24",
    });
  },
  formatDateMonth: (d: Date): string => {
    return d.toLocaleString(LOCALE, {
      day: "2-digit",
      month: "short",
    });
  },
  abbreviateCurrency: function (number: number) {
    return `${CURRENCY_SYMBOL}${this.abbreviateNumber(number)}`;
  },
  abbreviateNumber: (number: number) => {
    // what tier? (determines SI symbol)
    const tier = (Math.log10(Math.abs(number)) / 3) | 0;

    // if zero, we don't need a suffix
    if (tier == 0) return `${number}`;

    // get suffix and determine scale
    const suffix = SI_SYMBOL[tier];
    const scale = Math.pow(10, tier * 3);

    // scale the number
    const scaled = number / scale;

    // format number and add suffix
    return `${scaled.toFixed(1) + suffix}`;
  },
};
const SI_SYMBOL = ["", "k", "M", "B", "T", "P", "E"];

export default localeUtils;
