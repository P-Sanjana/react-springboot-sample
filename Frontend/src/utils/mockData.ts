import bitcoin from "../assets/coinImage/bitcoin.png";
import theme from "../theme/theme";

export const userProfile = {
  userId: 1,
  userFullname: "John",
  userAvatarUrl:
    "https://s3-alpha-sig.figma.com/img/973c/8b4c/05e06771ca57ec378072223b93fc6007?Expires=1635724800&Signature=LoSuQuAoiSGeFs04cT-kZICbfCNVFSd~jnfp6ffpCUDjKiWkq9dvi61je-5x7~toH3IMbNrCQrfctP-mrlVh~H7iAF-UlMNO66--TqlXExfb91MLGLnuK8O5HONWoVL0cT9Zxqx0iBMifwRo8oY~3f45YVs52Kw2FH8LwZxbYDhZCAev85Kqk7yofawgrWgZoYQJTaS~Re4TV09PF56F4wCcbZm81XaIgCbPDF2vZg~ybhPfMAeUf18TseFlsLPU1tbCigTOF3rPjeU5bSQvaBwJrDSGNUkeEdHxpfqfV4CihwWdxdXms67T5IcnBDlMR~AzVGJL~KIiu4lLZ~Fh5g__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA",
};

export const portfolioCards = [
  {
    icon: `${bitcoin}`,
    currencyType: "Bitcoin",
    currencyCode: "BTC",
    currencyValue: "$50000.00",
    gains: "+10.5",
    color: `${theme.palette.success.main}`,
  },
  {
    icon: `${bitcoin}`,
    currencyType: "Bitcoin",
    currencyCode: "BTC",
    currencyValue: "$50000.00",
    gains: "+10.5",
    color: `${theme.palette.success.main}`,
  },
  {
    icon: `${bitcoin}`,
    currencyType: "Bitcoin",
    currencyCode: "BTC",
    currencyValue: "$50000.00",
    gains: "+10.5",
    color: `${theme.palette.success.main}`,
  },
];

export const chips = [
  {
    name: "Bitcoin",
    color: "#F7931A",
    selected: true,
  },
  {
    name: "XRP",
    color: "#222222",
  },
  {
    name: "Polkadot",
    color: "#E6007A",
  },
  {
    name: "Ethereum",
    color: "#627EEA",
  },
  {
    name: "Tether",
    color: "#26A17B",
  },
  {
    name: "Ethereum 2",
    color: "#191971",
  },
  {
    name: "Dodge Coin",
    color: "#DBC984",
  },
];

export const data = {
  xAxisLabels: ["SEP 16", "SEP 24", "SEP 30", "OCT 07", "OCT 14"],
  dataArr: [
    {
      currencyColor: "#F7931A",
      currencyName: "Bitcoin",
      data: [100, 300, 250, 320, 400],
    },
    {
      currencyColor: "#0052FF",
      currencyName: "Total Investment",
      data: [50, 80, 76, 90, 150],
    },
  ],
};
