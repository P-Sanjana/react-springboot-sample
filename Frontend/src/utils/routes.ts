export const PURCHASE = "/purchase";
export const SELL = "/sell";
export const TRADE = "/trade";
export const DASHBOARD = "/";
export const WALLET = "/wallet";
export const DETAIL = "/detail/:id";
