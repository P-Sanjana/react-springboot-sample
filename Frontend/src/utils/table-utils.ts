import { Currency } from "../types/responseTypes";

export type SortDirection = "asc" | "desc" | "none";

export type SortEventHandler<T> = (
  fieldPicker: FieldPicker<T>,
  direction: SortDirection,
) => void;

export type FieldPicker<T> = (obj: T) => any;

type TableHeader<T> = {
  label: string;
  sort?: boolean;
  sortFieldPicker?: FieldPicker<T>;
};

export const tradeTableHeaders: Array<TableHeader<Currency>> = [
  {
    label: "Name",
  },
  {
    label: "Price",
  },
  {
    label: "Change",
  },
  {
    label: "MarketCap",
    sort: true,
    sortFieldPicker: (curr) => curr.marketCap,
  },
  {
    label: "Watch",
  },
];
