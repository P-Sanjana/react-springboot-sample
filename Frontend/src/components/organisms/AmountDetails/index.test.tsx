import React from "react";
import { render } from "@testing-library/react";
import AmountDetails from "./index";
import { selectedCrypto, balance } from "../../../utils/constants";

describe("Checkout Component", () => {
  it("should match snapshot", () => {
    const wrapper = render(
      <AmountDetails
        amountEventHandler={() => console.log("amount handler")}
        amountEntered={876400}
        selectedCrypto={selectedCrypto}
        convertedCoin={0.0346}
        convertedCoinHandler={() => console.log("converted")}
        buttonTitle="Buy Max"
        totalBal={balance}
        error=""
        setErrorHandler={() => console.log("error message")}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
