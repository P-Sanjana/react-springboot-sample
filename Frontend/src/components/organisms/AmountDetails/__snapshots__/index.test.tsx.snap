// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Amount details
        </p>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-firstCard-2 MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <div
              class="MuiInputBase-root MuiInputBase-adornedStart"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  $
                </h6>
              </div>
              <input
                class="MuiInputBase-input MuiInputBase-inputAdornedStart"
                placeholder="Enter Amount"
                type="text"
                value="876400"
              />
            </div>
            
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-7 MuiButton-outlinedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Buy Max
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-10 makeStyles-sliderCard-3"
        >
          <span
            class="MuiSlider-root MuiSlider-colorPrimary Mui-disabled MuiSlider-trackFalse MuiSlider-vertical"
          >
            <span
              class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-12"
            />
            <span
              class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-13"
              style="bottom: 0%; height: 50%;"
            />
            <input
              type="hidden"
              value="55"
            />
            <span
              aria-orientation="vertical"
              aria-valuemax="100"
              aria-valuemin="10"
              aria-valuenow="55"
              class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-11 MuiSlider-thumbColorPrimary Mui-disabled"
              data-index="0"
              role="slider"
              style="bottom: 50%;"
            />
          </span>
          <span
            class="MuiTypography-root makeStyles-bitcoinValue-6 MuiTypography-caption"
          >
            1
            BTC
            =
            $3,406,069.54
          </span>
        </div>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-secondCard-4 MuiPaper-outlined MuiPaper-rounded"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            0.0346
          </h6>
          <p
            class="MuiTypography-root makeStyles-BTCText-8 MuiTypography-body1"
          >
            BTC
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Amount details
      </p>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-firstCard-2 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <div
            class="MuiInputBase-root MuiInputBase-adornedStart"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1"
              >
                $
              </h6>
            </div>
            <input
              class="MuiInputBase-input MuiInputBase-inputAdornedStart"
              placeholder="Enter Amount"
              type="text"
              value="876400"
            />
          </div>
          
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-7 MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Buy Max
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-sliderCard-3"
      >
        <span
          class="MuiSlider-root MuiSlider-colorPrimary Mui-disabled MuiSlider-trackFalse MuiSlider-vertical"
        >
          <span
            class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-12"
          />
          <span
            class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-13"
            style="bottom: 0%; height: 50%;"
          />
          <input
            type="hidden"
            value="55"
          />
          <span
            aria-orientation="vertical"
            aria-valuemax="100"
            aria-valuemin="10"
            aria-valuenow="55"
            class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-11 MuiSlider-thumbColorPrimary Mui-disabled"
            data-index="0"
            role="slider"
            style="bottom: 50%;"
          />
        </span>
        <span
          class="MuiTypography-root makeStyles-bitcoinValue-6 MuiTypography-caption"
        >
          1
          BTC
          =
          $3,406,069.54
        </span>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-secondCard-4 MuiPaper-outlined MuiPaper-rounded"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          0.0346
        </h6>
        <p
          class="MuiTypography-root makeStyles-BTCText-8 MuiTypography-body1"
        >
          BTC
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
